### PR TITLE
BarChart on own analytics page stretched to 100% on y-axis

### LINKED ISSUE #53750: Stretch bar chart to 100% always on own analytics page

### PR DESCRIPTION
**Pull request by cbart**
Closes #53750 

## Test plan

Visual assessment and screenshots:

Same data: 1% codeowners and ownership overall.

Before:

Screenshot 2023-06-20 at 2 27 26 PM

After:

Screenshot 2023-06-20 at 10 44 39 AM

### CODE REVIEW

File: client/wildcard/src/components/Charts/components/bar-chart/BarChart.tsx [hunk 1]
```diff
@@ -76,13 +78,12 @@ export function BarChart<Datum>(props: BarChartProps<Datum>): ReactElement {
...
-                domain: [0, Math.max(...categories.map(category => category.maxValue))],
-            }),
-        [categories]
-    )
+    const yScale = useMemo(() => {
+        const highestValue = Math.max(...categories.map(category => category.maxValue))
+        return scaleLinear<number>({
+            domain: [0, Math.max(highestValue, maxValueLowerBound ?? -Infinity)],
+        })
+    }, [categories])
```
Comment (fkling) [inline client/wildcard/src/components/Charts/components/bar-chart/BarChart.tsx:86]: I think the dependency array is missing `maxValueLowerBound`. Also there is a maximum number of parameters a function can take... if `categories` will ever be larger than that the call will fail. I don't know the maximum off the top of my head, and I would assume that we have other problems if `categories` will ever be that size, but who knows.

As an alternative, something like (needs testing)

```suggestion
    const yScale = useMemo(() => scaleLinear<number>({
        domain: [0, categories.reduce((max, category) => Math.max(max, category.maxValue), maxValueLowerBound ?? -Infinity)],
    }), [categories, maxValueLowerBound])
```

Comment (cbart) [reply client/wildcard/src/components/Charts/components/bar-chart/BarChart.tsx]: Thank you!

**Comment by cbart** [comment]
Thank you @fkling. Can confirm it still renders OK.

**Review by fkling — APPROVED** [review]

**Review by vovakulikov — APPROVED** [review]
👍 , I found that `maxValueLowerBound` sounds a bit difficult to understand, maybe we could add a line or two of JSDoc comment with a brief description of this setting

**Pull request merged**